core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/CryptoUtil.java

```diff
@@ -441,5 +445,21 @@ private String calculateThumbprint(Certificate certificate, String digest)
 
         return strBuffer.toString();
     }
+
+    private String getDefaultEncryptionAlgorithm(boolean isSymmetricKeyEncryptionEnabled) {
+
+        if (isSymmetricKeyEncryptionEnabled) {
+            return DEFAULT_SYMMETRIC_CRYPTO_ALGORITHM;
+        } else {
+            // Set the default crypto algorithm.
+            return DEFAULT_CRYPTO_ALGORITHM;
+        }
+    }
+
+    private boolean isSymmetricKeyEncryptionEnabled() {
+
+        return SYMMETRIC_INTERNAL_CRYPTO_PROVIDER_CLASS_NAME
+                .equals(serverConfigService.getFirstProperty(INTERNAL_CRYPTO_PROVIDER));
+    }
```

Comment:
What if this method is utilized in 'getDefaultEncryptionAlgorithm' rather than passing an additional boolean

Reply:
Since we use this variable at https://github.com/wso2/carbon-kernel/pull/2636/files#diff-8b0422c9fd8f481b4c8261829499efe4R161 also, I defined it seperately. If use this inside the getDefaultEncryptionAlgorithm method, it is not accessible in https://github.com/wso2/carbon-kernel/pull/2636/files#diff-8b0422c9fd8f481b4c8261829499efe4R161.
We can declare an instance variable and assign to it in getDefaultEncryptionAlgorithm also. Instead I went with this approach.